### DISCUSSION TITLE
Hide creation button when not having creation rights

### DISCUSSION THREAD

**Discussion by DBouwmans** [apache/superset#18516] [Ideas]
**Problem description**
As a user of superset, with rights to only view existing dashboards, I would expect not to see the + dashboard buttons. Currently, the + dashboard button provides an error that the user doesn’t have the rights to create a dashboard, but I want to hide the button instead of providing this error. 

This is exactly the same for creating dashboards and queries. I don’t want the ReadOnly end users to see the creation buttons. The fact that they can click on it makes it confusing to them.

**Current situation**
-	Still able to click on add dashboard/chart/SQL query when they don’t have the rights to do so. When they do, the following error ‘Access is Denied’ appears
   
![image](https://user-images.githubusercontent.com/71815117/137145790-4521a09d-358d-4340-981a-346c01c8dd1e.png)
![image](https://user-images.githubusercontent.com/71815117/137145805-f2da14e4-3915-4b4a-9806-2388ea8ccda0.png)

-	Same applies to the + buttons on the main page. 
-	The user is still able to view the ‘Mine’ tabs (which is unnecessary, since they can’t have dashboards of their own since they don’t have the rights) 
 
![image](https://user-images.githubusercontent.com/71815117/137145822-f2fd3f37-6c86-4d8f-ad4d-b56be1e09bb0.png)

**Desired solution**
- There are no + buttons for creation actions a user doesn’t have the rights for (in the menu bar or on the home page) 
- The ‘Mine’ tabs are no longer visible when the user doesn’t have the rights for creating that object. 
- For queries, the whole part of saved queries could be deleted since you can only view your own. However, I know that a functionality to favorite queries is requested/being developed so I would understand the choice to not do this.

**Describe alternatives you've considered**
This couldn’t be successfully resolved in css because it only relates to specific roles, others should still be able to view the + Dashboard button when they do have a role with these rights. 

**Additional context**
Please let me know if this feature request requires any additional information

**Comment by rosemarie-chiu** [comment]
I believe this PR: https://github.com/apache/superset/pull/17157 help solved the top navigation + button part. cc @hughhhh
